### DISCUSSION TITLE
Find and Replace Highlighting

### DISCUSSION THREAD

**Discussion by Patrock404** [olimorris/onedarkpro.nvim#212] [Q&A] [answered]
Does anyone know how to specify a color to use for the current word of a find and replace command?

For example, if I'm finding and replacing 'local' with 'test' in this file and confirming each one, how can I highlight the current 'local' that is waiting for confirmation?

![image](https://github.com/olimorris/onedarkpro.nvim/assets/113575251/8a04ffac-ce59-4185-952f-b313cd600d7a)

**Comment by mmirus** [comment, marked as answer]
TLDR: try configuring the `IncSearch` highlight group.

---

:wave: I did a little testing and I think this is what I'm seeing:

**If you're doing a search**

Example: `/local`

- While you're typing the search, the `IncSearch` highlight group controls the appearance of the next match
- `Search` controls the remaining matches while typing, and all matches after you press enter

**If you're doing a substitute without confirm enabled**

Example: `%s/local/test/`

- `Substitute` controls the appearance of both matches (while typing that part) and substitutions (while/after typing that part)

**If you're doing a substitute with confirm enabled**

Example: `:s/local/test/gc` or `:%s/local/test/c`

- `Substitute` applies as above while you're typing the command
- After you hit Enter, `IncSearch` controls the appearance of the match that is currently pending confirmation, and `Search` controls the appearance of the remaining matches

It was a little tricky to suss this out, but I think that's accurate.

**Reply by Patrock404** [reply]
You were right! After some fiddling around with colors, I got it to behave like I wanted. 

I just had to set the background and foregroud for the `IncSearch` highlight group. 

Thank you for your help!

![image](https://github.com/olimorris/onedarkpro.nvim/assets/113575251/0043f0c1-e2d0-447c-8796-9be84908f267)
